https://issues.redhat.com/browse/ACM-29169-tech review ready -new fine grain and reorg … (#8606)
open-cluster-management/rhacm-docs: secure_clusters/fine_grain_rbac_enable.adoc

[#fine-grain-enable]
= Enabling fine-grained role-based access control for virtualization 

{acm} supports fine-grained role-based access control (RBAC) for virtual machine scenarios. As a cluster administrator, you can manage and control permissions at the namespace level and cluster level on your managed clusters. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster.

Enabling fine-grained RBAC significantly changes how the console displays search results and manages access for application lifecycles, governance, and observability. 

*Required access:* Cluster administrator

To learn about {ocp-short} default and virtualization roles and permissions, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/virtualization/about#authorization_virt-security-policies[Authorization] in the {ocp-short} documentation.

See xref:../secure_clusters/rbac_implement_rhacm.adoc#rhacm-rbac-implement[Implementing role-based access control] for more details about {acm-short} role-based access.

> .Prerequisites

Author: let's discuss these prereqs--old and new.

Author: Done and changed

[#fine-grain-enable]

> = Enabling fine-grained role-based access control for virtualization 

Author: New content

> Enable fine-grained role-based access control for virtualization to manage and grant permissions to other users at the namespace level and cluster level on your managed clusters. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster. 

Reviewer: ```suggestion
Enable fine-grained role-based access control for virtualization to manage and grant permissions to other users at the namespace level and cluster level on your managed clusters. Allows the ability to grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster. 
```

small change just to make sure the customer knows they can do both namespace and cluster level

Reviewer: i think everything else is good just this small suggestion then i can approve the pr

> Enable fine-grained role-based access control for virtualization to manage and grant permissions to other users at the namespace level and cluster level on your managed clusters. Allows the ability to grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster. 

Reviewer 2: Seems wordy and there is a fragment...."Allows the ability to..."

```suggestion
Enable fine-grained role-based access control for virtualization to manage and grant permissions to other users at the namespace level and cluster level on your managed clusters. RBAC for virtualization provides the capability to grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster. 
```

Author: We need to keep _permission_ on this one. Additionally the developer asked me to add to _other users_. The fragment is correct (good catch! TY), but other changes I do not feel are necessary, as they wanted to use "permission". This was not incorrect, so I would like to keep it. See the edits and let me know if that works.

*Required access:* Cluster administrator

To learn about {ocp-short} default and virtualization roles and permissions, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/virtualization/about#authorization_virt-security-policies[Authorization] in the {ocp-short} documentation.

.Prerequisites

See the following prerequisites that are needed to enable fine-grained role-based access control for virtual machines in your environment:

> - Install a supported version of {ocp-virt} on your hub cluster on the hub cluster and all managed clusters where you intend to manage virtual machines. For more information, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/virtualization/installing[Virtualization - Installing].

Reviewer 2: ```suggestion
- Install a supported version of {ocp-virt} on your hub cluster and all managed clusters where you intend to manage virtual machines. For more information, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/virtualization/installing[Virtualization - Installing].
```

Author: better to use "your" so let's keep that. We use your in the same line so it agrees.

- *Optional:* Install the {mtv} if you plan to perform virtual migrations. See link:../virtualization/migrate_vm.adoc#migrate-vm[Migrating virtual machines between clusters].

- *Note:* If you enable the `cnv-mtv-integrations` feature in {acm-short}, the product automatically installs and configures both the {ocp-virt} and {mtv} for you.

- Ensure your hub cluster is self-managed as a `local-cluster`, which means the `disableHubSelfManagement` parameter is set to `false`, the default setting from when you installed {acm-short}.

> - Configure the {acm-short} hub cluster and all managed clusters with the same identities. Users, groups, and group memberships must be identical across the environment to ensure that an identity authenticated on the hub cluster maps correctly to the same identity on the managed clusters. (link to the OCP feature)

Reviewer 2: Which OCP feature?

Reviewer 3: We don't need an OCP feature link here. We are giving guidance on what ACM requires, and we are letting customer figure out how to get there.

Author: Just forgot to remove that after our meeting. Good catch.

Author: This is removed.

.Procedure 

See the following steps to enable the `fine-grained-rbac` component in the `MultiClusterHub` resource:

> . Run the following command to patch the resource. The default name of the custom resource is `multiclusterhub`, and `open-cluster-management` is the default namespace: If you have a different name or namespace, update the command: 

Reviewer 2: ```suggestion
. Run the following command to patch the resource. The default name of the custom resource is `multiclusterhub`, and `open-cluster-management` is the default namespace. If you have a different name or namespace, update the command: 
```

[#fine-grain-enable]
= Enabling fine-grained role-based access control for virtualization 

> {acm} supports fine-grained role-based access control (RBAC) for virtual machine scenarios. As a cluster administrator, you can manage and control permissions at the namespace level and cluster level on your managed clusters. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster.

Reviewer: I would like to combine the last two sentences here and say that you can manage and grant permissions to other users.

the idea would be that as the super top admin, they can give permissions to others at namespace or cluster level

Author: We don't want to necessarlity combine because it would be too long. It's good to have a couple sentences rather than one long one the the short description. I have made a change to the line before based on your feedback. If the following line is still true, I would rather let it stand alone if that is OK:

Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster.

> {acm} supports fine-grained role-based access control (RBAC) for virtual machine scenarios. As a cluster administrator, you can manage and grant permissions to other users at the namespace level and cluster level on your managed clusters. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster.

Reviewer 3: Duplicate line mentioned in intro section. We can talk about it in the meeting tomorrow.

> Enabling fine-grained RBAC significantly changes how the console displays search results and manages access for application lifecycles, governance, and observability. 

Reviewer: @mshort55 any thoughts on this? I wasnt as involved with other this affected other areas but i was aware there were some search issues that happened when enabling fine-grained rbac

Author: Also content that was there already...we can remove if needed.

Reviewer 3: I think we should remove this, or at the very least reword it. We fixed the main bugs that caused issues/regressions with search when fine grained rbac was enabled. We more or less have backwards compatibility with old ACM rbac behavior and the new fine grained rbac behavior. By saying this, I think we will scare people and get messages telling us to explain this part.

Author: remove this

*Required access:* Cluster administrator

To learn about {ocp-short} default and virtualization roles and permissions, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/virtualization/about#authorization_virt-security-policies[Authorization] in the {ocp-short} documentation.

.Prerequisites

> See the following prerequisites that are needed to enable fine-grained role-based access control:

Reviewer: seems like there is a request to add virtualization wording here see the google doc
https://docs.google.com/document/d/1OiP9vx7xMYkMPwTjSV5rofEgNAy__dW46Qrl2pFpuCs/edit?disco=AAAByzuBN4k

Author: We add it in the title and the description. Added to this line, as well:

See the following prerequisites that are needed to enable fine-grained role-based access control for virtual machines in your environment:

- Install a supported version of {ocp-virt} on your hub cluster on the hub cluster and all managed clusters where you intend to manage virtual machines. For more information, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/virtualization/installing[Virtualization - Installing].

- *Optional:* Install the _Migration Toolkit for Virtualization operator_ if you plan to perform virtual migrations. (link--= Migrating virtual machines between clusters then check the OCP link))

> - *Note:* If you enable the `cnv-mtv-integrations` feature in {acm-short}, the system automatically installs and configures both the OpenShift Virtualization and MTV operators for you.

Reviewer: can we replace system with acm?

https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.15/html/virtualization/acm-virt#migrate-vm
can we link to this doc about enabling because there are a bit more steps involve to enable it correctly so i dont want it to sound misleading when mentioning automatically

- Ensure your hub cluster is self-managed as a `local-cluster`, which means the `disableHubSelfManagement` parameter is set to `false`, the default setting from when you installed {acm-short}.

> - Configure the {acm-short} hub cluster and all managed clusters with the same identity providers (IdP). Users, groups, and group memberships must be identical across the environment to ensure that an identity authenticated on the hub cluster maps correctly to the same identity on the managed clusters. (link to the OCP feature)

Reviewer 3: ```suggestion
- Configure the {acm-short} hub cluster and all managed clusters with the same identities. Users, groups, and group memberships must be identical across the environment to ensure that an identity authenticated on the hub cluster maps correctly to the same identity on the managed clusters. (link to the OCP feature)
```
Per Joydeep we are not saying IDP must be the same anymore. For now we let customer figure out the IDP part and we just say that identities must match between all clusters.

.Procedure 

See the following steps to enable the `fine-grained-rbac` component in the `MultiClusterHub` resource:

. Run the following command to patch the resource. The default name of the custom resource is `multiclusterhub`, and `open-cluster-management` is the default namespace: If you have a different name or namespace, update the command: 

> //can this be changed?

Reviewer: this whole part `Run the following command to patch the resource. The default name of the custom resource is `multiclusterhub`, and `open-cluster-management` is the default namespace: If you have a different name or namespace, update the command: `
can be removed in my opinion if there is a conventional way it is doc to enabling a feature
I was looking at others and it seems like just providing the oc command works?

what do you think @swopebrandi1

Author: We can if the users are not changing the namespace. I am not sure I have ever had to mention the default _resource name_ bc I didn't think that changed. I think that part was in the draft.

My comment on line 31 was inquiring about that...do we need to say this.

(other doc only provides the command and doesn't mention what can or may change about it because the writer may not have asked the proper questions or knew to ask, or there may be no need to mention it)

Author: > this whole part `Run the following command to patch the resource. The default name of the custom resource is `multiclusterhub`, and `open-cluster-management`is the default namespace: If you have a different name or namespace, update the command:` can be removed in my opinion if there is a conventional way it is doc to enabling a feature I was looking at others and it seems like just providing the oc command works?
> 
> what do you think [@swopebrandi1](https://github.com/swopebrandi1)

We can if the users are not changing the namespace. I am not sure I have ever had to mention the default _resource_ name bc I didn't think that changed. I think that part was in the draft.

My comment on line 31 was inquiring about that...do we need to say this?

(other doc only provides the command and doesn't mention what can or may change about it because the writer may not have asked the proper questions or knew to ask, or there may be no need to mention it--but if the user can and would change the namespace, we should say it is default.)

[source,bash]
----
oc patch mch multiclusterhub -n open-cluster-management --type=json -p='[{"op": "add", "path": "/spec/overrides/components/-", "value": {"name": "fine-grained-rbac", "enabled": true}}]'
----

Apply the `environment=virtualization` label to only the `local-cluster` to enable the necessary policies and RBAC configurations. Run the following command:

[source,bash]
----
oc label managedcluster local-cluster environment=virtualization

> ----

Reviewer 3: ```suggestion
```
This section is no longer needed. We removed this requirement with a recent update for 2.16 to improve customer experience.
